agent/connect/ca/provider_consul.go

```diff
@@ -447,6 +429,8 @@ func (c *ConsulProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 // are met. It should return a signed CA certificate with a path length constraint
 // of 0 to ensure that the certificate cannot be used to generate further CA certs.
 func (c *ConsulProvider) SignIntermediate(csr *x509.CertificateRequest) (string, error) {
+	connect.HackSANExtensionForCSR(csr)
```

Comment:
Do we need this hack if the CommonName isn't being removed from intermediates?

Reply:
You have a point. I blanket added this before any of the relevant `crypto/x509` library calls in the spirit of something approximating monkeypatching. I also did all of this while trying to figure out why nothing was working after getting fixed (which was an unrelated bug that I later fixed). I did not revisit to see of the intermediates actually had this hack *activate* practically speaking.

Reply:
Tested removing this call from SignIntermediate functions in the providers and everything (even AWS) works.